### PR TITLE
Added KeyTable.order_by

### PR DESCRIPTION
**Pull request by cseed**
Added Type.ordering.
Added KeyTable.from_py.

### CODE REVIEW

File: src/main/scala/is/hail/variant/Variant.scala [hunk 1]
```diff
@@ -124,10 +128,17 @@ case class AltAllele(ref: String,
...
   def toJSON: JValue = JObject(
     ("ref", JString(ref)),
-    ("alt", JString(alt))
-  )
+    ("alt", JString(alt)))
 
   override def toString: String = s"$ref/$alt"
+
+  def compare(that: AltAllele): Int = {
+    var c = ref.compare(that.ref)
```
Comment (tpoterba) [inline src/main/scala/is/hail/variant/Variant.scala]: can be a val

File: src/main/scala/is/hail/utils/package.scala [hunk 2]
```diff
@@ -226,13 +226,21 @@ package object utils extends Logging
 
   def uriPath(uri: String): String = new URI(uri).getPath
 
-  def extendOrderingToNull[T](implicit ord: Ordering[T]): Ordering[Any] = {
+  def extendOrderingToNull[T](missingGreatest: Boolean)(implicit ord: Ordering[T]): Ordering[Any] = {
     new Ordering[Any] {
-      def compare(a: Any, b: Any) =
+      def compare(a: Any, b: Any): Int =
         (a, b) match {
```
Comment (tpoterba) [inline src/main/scala/is/hail/utils/package.scala]: is it worth rewriting this to avoid allocation?

Comment (cseed) [inline src/main/scala/is/hail/utils/package.scala]: Probably not, but I did anyway.

**Comment by cseed** [comment]
Addressed comments.

**Pull request merged**